src/services/event.ts

```diff
@@ -273,14 +301,29 @@ export async function updateEvent(id: number, request: UpdateEventRequest): Prom
     data,
   } = request;
 
-  // Get current json owner.
-  const { owner } = event.data;
-  // if owner changes update the json owner.
-  if (owner && ownerId !== event.data.owner.id) {
-    // get the new owner.
-    const newOwner = await User.findByPk(ownerId, { attributes: ['id', 'name', 'email'], raw: true });
-    // update the owner in the data.
-    data.owner = newOwner;
+  if (ownerId) {
+    const newOwner = await User.findOne(
+      {
+        where: { id: ownerId },
+        attributes: [
+          'id',
+          'nameWithNationalCenters',
+          'email',
+          'name',
+        ],
+        include: [
+          {
+            model: db.NationalCenter,
```

Comment:
does this need attributes: []

Reply:
I don't believe the virtual field works with attributes: []